### DISCUSSION TITLE
password props on in input widget

### DISCUSSION THREAD

**Discussion by mlafois** [Textualize/textual#905] [Q&A] [answered]
Hello,

I'm playing with input's widget from the css branch.

`Input(placeholder="Password", password=True),`

This code raise an error : 

`TypeError: Input.__init__() got an unexpected keyword argument 'password'`

I have manage this by adding the hereunder lines in _input.py

` def __init__(
        self,
        value: str = "",
        placeholder: str = "",
        highlighter: Highlighter | None = None,
        name: str | None = None,
        id: str | None = None,
        classes: str | None = None,
        password: bool = False, # Here
    ) -> None:
        super().__init__(name=name, id=id, classes=classes)
        self.value = value
        self.placeholder = placeholder
        self.highlighter = highlighter
        self.password = password` # Here

**Comment by davep** [comment, marked as answer]
Looks fair to me to expect that to be available via a keyword argument (I don't think it was documented anywhere in the CSS docs to be one, only documented as being a reactive property); I've created a PR for the rest of the team to review: https://github.com/Textualize/textual/pull/911

**Reply by mlafois** [reply]
Sorry you're right I have seen it from the gh-pages (widgets/input/#reactive-attributes). It is because Reactive property still not clear to me.

**Reply by davep** [reply]
No worries; the docs are still in a bit of a state of flux as the CSS branch heads towards release, so I imagine some things might not be 100% clear.

For what it's worth, a reactive property is just like any other property of a class, just with some extra "magic" (which you can see documented in the CSS docs if you're reading them). But, as with any Python class, it isn't the case that every property defined by a class has a corresponding argument in the constructor. So, with `Input`, you could have instantiated it then set `password` something liken this:

```python
pswd = Input(placeholder="Password")
pswd.password = True
```

and then used `pswd` in your wider code. But personally I feel it makes sense that it's added as a keyword argument.

**Reply by mlafois** [reply]
Ok thank for the explanation it's more clear to me

**Comment by davep** [comment]
The PR (https://github.com/Textualize/textual/pull/911) is now approved and merged; `password` will be an available keyword parameter for `Input` once the CSS branch of Textual is released. Thanks for suggesting it. :-)
